### PR TITLE
Some CI improvements

### PR DESCRIPTION
**Pull request by giordano**
* Add CompatHelper
* Reduce frequency of TagBot to once per day
* Remove Cirrus script, it's not working properly lately
* Add FreeBSD to Travis
* Run Travis only on master and a few special-named branches
* Cache artifacts in Travis and AppVeyor

### CODE REVIEW

**Comment by giordano** [comment]
Is there an easy way to exclude FreeBSD with MKL?

**Comment by ararslan** [comment]
> Is there an easy way to exclude FreeBSD with MKL?

Should be able to use `exclude:` in `jobs:`, I believe.

**Comment by giordano** [comment]
> Should be able to use exclude: in jobs:, I believe.

Does it work with environment variables?  I didn't try.  I went with the opposite way: `include`-ing the FreeBSD jobs :sweat_smile:

**Comment by ararslan** [comment]
Yeah I believe it does.

**Comment by giordano** [comment]
Looks like it doesn't?

**Comment by giordano** [comment]
Shall I go back to `include`-ing or maybe I did something wrong now?

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,22 @@ notifications:
...
+
+cache:
+  directories:
+    - $HOME/.julia/artifacts
+
 jobs:
+  exclude:
+    - os: freebsd
+      env:
+        - JULIA_FFTW_PROVIDER=MKL
```
Comment (ararslan) [inline .travis.yml:39]:
```suggestion
      env: JULIA_FFTW_PROVIDER=MKL
```
? Not sure it will make a difference but worth a try I guess

Comment (giordano) [reply .travis.yml]: :open_mouth:

**Comment by ararslan** [comment]
Looks like that worked! :tada:

**Pull request merged**